snerk.py

```diff
@@ -501,8 +501,7 @@ def privmsg(self, sender, channel, msg):
             self.msg(channel, random.choice(sayings))
             return
 
-        elif (re.search(r'^\s*!\s*logs?\s*$', msg, re.I|re.U) or
-              re.search(r'^\s*logs?\s*!\s*$', msg, re.I|re.U)):
+        elif (re.search(r'(?<=\W)!logs?(?=\W)|(?<=\W)logs?!(?=\W)'ig, msg, re.I|re.U)
```

Comment:
I very much doubt this line works currently. If nothing else, the mandatory trailing colon is missing. I don't know what you were going for with the `'ig` part.

Maybe something like `elif re.search(r'^([ !?]*logs?|logs?[ !?]*)$', msg, re.I|re.U):` would work (untested)? I'm still not sure of your exact requirements.